{
    "dataType": "CVE_RECORD",
    "dataVersion": "5.0",
    "cveMetadata": {
        "cveId": "CVE-2024-0960",
        "assignerOrgId": "1af790b2-7ee1-4545-860a-a788eba489b5",
        "state": "PUBLISHED",
        "assignerShortName": "VulDB",
        "dateReserved": "2024-01-26T17:19:38.643Z",
        "datePublished": "2024-01-27T12:00:07.334Z",
        "dateUpdated": "2024-02-09T19:12:52.566Z"
    },
    "containers": {
        "cna": {
            "providerMetadata": {
                "orgId": "1af790b2-7ee1-4545-860a-a788eba489b5",
                "shortName": "VulDB",
                "dateUpdated": "2024-02-09T19:12:52.566Z"
            },
            "title": "flink-extended ai-flow workflow_command.py cloudpickle.loads deserialization",
            "problemTypes": [
                {
                    "descriptions": [
                        {
                            "type": "CWE",
                            "cweId": "CWE-502",
                            "lang": "en",
                            "description": "CWE-502 Deserialization"
                        }
                    ]
                }
            ],
            "affected": [
                {
                    "vendor": "flink-extended",
                    "product": "ai-flow",
                    "versions": [
                        {
                            "version": "0.3.1",
                            "status": "affected"
                        }
                    ]
                }
            ],
            "descriptions": [
                {
                    "lang": "en",
                    "value": "A vulnerability was found in flink-extended ai-flow 0.3.1. It has been declared as critical. Affected by this vulnerability is the function cloudpickle.loads of the file \\ai_flow\\cli\\commands\\workflow_command.py. The manipulation leads to deserialization. The attack can be launched remotely. The complexity of an attack is rather high. The exploitation appears to be difficult. The exploit has been disclosed to the public and may be used. The identifier VDB-252205 was assigned to this vulnerability."
                },
                {
                    "lang": "de",
                    "value": "In flink-extended ai-flow 0.3.1 wurde eine kritische Schwachstelle ausgemacht. Es geht um die Funktion cloudpickle.loads der Datei \\ai_flow\\cli\\commands\\workflow_command.py. Durch Manipulieren mit unbekannten Daten kann eine deserialization-Schwachstelle ausgenutzt werden. Der Angriff kann über das Netzwerk erfolgen. Die Komplexität eines Angriffs ist eher hoch. Sie gilt als schwierig ausnutzbar. Der Exploit steht zur öffentlichen Verfügung."
                }
            ],
            "metrics": [
                {
                    "cvssV3_1": {
                        "version": "3.1",
                        "baseScore": 5,
                        "vectorString": "CVSS:3.1/AV:N/AC:H/PR:N/UI:R/S:U/C:L/I:L/A:L",
                        "baseSeverity": "MEDIUM"
                    }
                },
                {
                    "cvssV3_0": {
                        "version": "3.0",
                        "baseScore": 5,
                        "vectorString": "CVSS:3.0/AV:N/AC:H/PR:N/UI:R/S:U/C:L/I:L/A:L",
                        "baseSeverity": "MEDIUM"
                    }
                },
                {
                    "cvssV2_0": {
                        "version": "2.0",
                        "baseScore": 5.1,
                        "vectorString": "AV:N/AC:H/Au:N/C:P/I:P/A:P"
                    }
                }
            ],
            "timeline": [
                {
                    "time": "2024-01-26T00:00:00.000Z",
                    "lang": "en",
                    "value": "Advisory disclosed"
                },
                {
                    "time": "2024-01-26T01:00:00.000Z",
                    "lang": "en",
                    "value": "VulDB entry created"
                },
                {
                    "time": "2024-01-26T18:24:44.000Z",
                    "lang": "en",
                    "value": "VulDB entry last update"
                }
            ],
            "credits": [
                {
                    "lang": "en",
                    "value": "bayuncao (VulDB User)",
                    "type": "reporter"
                }
            ],
            "references": [
                {
                    "url": "https://vuldb.com/?id.252205",
                    "tags": [
                        "vdb-entry",
                        "technical-description"
                    ]
                },
                {
                    "url": "https://vuldb.com/?ctiid.252205",
                    "tags": [
                        "signature",
                        "permissions-required"
                    ]
                },
                {
                    "url": "https://github.com/bayuncao/vul-cve-8",
                    "tags": [
                        "broken-link"
                    ]
                },
                {
                    "url": "https://github.com/bayuncao/vul-cve-8/blob/main/dataset.pkl",
                    "tags": [
                        "broken-link",
                        "exploit"
                    ]
                }
            ]
        }
    }
}